crates/pop-contracts/src/utils/metadata.rs

```diff
@@ -0,0 +1,367 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use crate::errors::Error;
+use contract_extrinsics::ContractArtifacts;
+use contract_transcode::ink_metadata::MessageParamSpec;
+use scale_info::form::PortableForm;
+use std::path::Path;
+
+#[derive(Clone, PartialEq, Eq)]
+/// Describes a contract message.
+pub struct Param {
+	/// The label of the parameter.
+	pub label: String,
+	/// The type name of the parameter.
+	pub type_name: String,
+}
+
+#[derive(Clone, PartialEq, Eq)]
+/// Describes a contract function.
+pub struct ContractFunction {
+	/// The label of the constructor.
+	pub label: String,
+	/// If the message accepts any `value` from the caller.
+	pub payable: bool,
+	/// The parameters of the deployment handler.
+	pub args: Vec<Param>,
+	/// The constructor documentation.
+	pub docs: String,
+	/// If the constructor is the default for off-chain consumers (e.g UIs).
+	pub default: bool,
```

Comment:
As this only applies to constructors, might it be better to move these field definitions into the enum variants instead?

Reply:
`default` exist in messages, let me change the docs. The only only applies to messages (and not `contructor`) is mutates.